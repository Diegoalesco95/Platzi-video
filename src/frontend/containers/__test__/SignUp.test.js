import React from 'react';
import { create } from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderMock from '../../__mocks__/ProviderMock';
import SignUp from '../Signup';

configure({ adapter: new Adapter() });

describe('Header component', () => {
  test('Match Snapshot', () => {
    const signup = create(
      <ProviderMock>
        <SignUp />
      </ProviderMock>,
    );
    expect(signup.toJSON()).toMatchSnapshot();
  });

  it('Calls and executes preventDefault function onSubmit form', () => {
    const preventDefault = jest.fn();
    const wrapper = mount(
      <ProviderMock>
        <SignUp />
      </ProviderMock>,
    );
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    wrapper.unmount();
  });
});
