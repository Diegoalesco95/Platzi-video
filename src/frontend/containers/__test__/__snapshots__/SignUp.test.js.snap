// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Match Snapshot 1`] = `
Array [
  <header
    className="header isSignUp"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo Platzi Video"
        className="header__img"
        src="test-file.stubs"
      />
    </a>
    <div
      className="header__menu"
    >
      <div
        className="header__menu--profile"
      >
        <img
          alt="user Icon"
          src="test-file.stubs"
        />
        <p>
          Menu
        </p>
      </div>
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Iniciar Sesión
          </a>
        </li>
        <li>
          <a
            href="/signUp"
            onClick={[Function]}
          >
            Registrarse
          </a>
        </li>
      </ul>
    </div>
  </header>,
  <section
    className="register"
  >
    <section
      className="register__container"
    >
      <h2>
        Regístrate
      </h2>
      <form
        className="register__container--form"
        onSubmit={[Function]}
      >
        <input
          aria-label="Nombre"
          className="input--signup"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          required={true}
          type="text"
        />
        <input
          aria-label="Correo"
          className="input--signup"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Correo"
          required={true}
          type="email"
        />
        <input
          aria-label="Contraseña"
          className="input--signup"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Contraseña"
          required={true}
          type="password"
        />
        <input
          aria-label="Confirmar contraseña"
          className="input--signup"
          id="confirm-password"
          name="password"
          onChange={[Function]}
          placeholder="Confirmar contraseña"
          required={true}
          type="password"
        />
        <button
          className="button"
          type="submit"
        >
          Registrarme
        </button>
      </form>
      <section
        className="register__container--social-media"
      >
        <div
          className="google"
        >
          <img
            alt="Google"
            className="google__img"
            src="test-file.stubs"
          />
          <p>
            <a
              href="/"
            >
              Registro con Google 
            </a>
          </p>
        </div>
        <div
          className="twitter"
        >
          <img
            alt="Twitter"
            className="twitter__img"
            src="test-file.stubs"
          />
          <p>
            <a
              href="/"
            >
              Registro con Twitter
            </a>
          </p>
        </div>
      </section>
      <p
        className="register__container--login"
      >
        ¿Ya tienes una cuenta?
        <br />
        <a
          href="/login"
          onClick={[Function]}
        >
          Inicia sesión
        </a>
      </p>
    </section>
  </section>,
]
`;
